### PR TITLE
Add  magnificationScale to CupertinoMagnifier for Zoom Effect

### LINKED ISSUE #155275: The CupertinoMagnifier has no magnification Scale attribute
### Steps to reproduce

1. Run the code sample
2. Drag on the Flutter logo

### Expected results

The expected result is the zoom effect inside the `CupertinoMagnifier` area.

### Actual results

The `CupertinoMagnifier` widget is not displaying the expected zoom effect. 
This issue arises because the widget internally wraps the `RawMagnifier`, which has its `magnificationScale` attribute set to the default value of 1. As a result, no magnification is applied, and the zoom effect is absent.

[Docs:](https://api.flutter.dev/flutter/cupertino/CupertinoMagnifier-class.html)
> [CupertinoMagnifier](https://api.flutter.dev/flutter/cupertino/CupertinoMagnifier-class.html) is a wrapper around [RawMagnifier](https://api.flutter.dev/flutter/widgets/RawMagnifier-class.html) that handles styling and transitions.

### Code sample

<details open><summary>Code sample</summary>

```dart
import 'package:flutter/cupertino.dart';
import 'package:flutter/material.dart';

void main() => runApp(const CupertinoMagnifierApp());

class CupertinoMagnifierApp extends StatelessWidget {
  const CupertinoMagnifierApp({super.key});

  @override
  Widget build(BuildContext context) {
    return const CupertinoApp(
      theme: CupertinoThemeData(brightness: Brightness.light),
      home: CupertinoMagnifierExample(),
    );
  }
}

class CupertinoMagnifierExample extends StatefulWidget {
  const CupertinoMagnifierExample({super.key});

  @override
  State<CupertinoMagnifierExample> createState() => _CupertinoMagnifierExampleState();
}

class _CupertinoMagnifierExampleState extends State<CupertinoMagnifierExample> {
  static const double magnifierRadius = 50.0;
  Offset dragGesturePosition = Offset.zero;

  @override
  Widget build(BuildContext context) {
    return  CupertinoPageScaffold(
      navigationBar: const CupertinoNavigationBar(
        middle: Text('CupertinoMagnifier Sample'),
      ),
      child: Center(
          child: Column(
            mainAxisAlignment: MainAxisAlignment.center,
            children: <Widget>[
               const Text('Drag on the logo!'),
              RepaintBoundary(
                child: Stack(
                  children: <Widget>[
                    GestureDetector(
                      onPanUpdate: (DragUpdateDetails details) => setState(
                        () {
                          dragGesturePosition = details.localPosition;
                        },
                      ),
                      onPanDown: (DragDownDetails details) => setState(
                        () {
                          dragGesturePosition = details.localPosition;
                        },
                      ),
                      child: const FlutterLogo(size: 200),
                    ),
                    Positioned(
                      left: dragGesturePosition.dx - magnifierRadius,
                      top: dragGesturePosition.dy - magnifierRadius,
                      child: const CupertinoMagnifier(
                        borderRadius: BorderRadius.all(Radius.circular(magnifierRadius)),
                        additionalFocalPointOffset: Offset(0, -magnifierRadius),
                      ),
                    )
                  ],
                ),
              ),
            ],
          ),
        ),
    );
  }
}

```

</details>

### Screenshots or Video

<details open>
<summary>Screenshots / Video demonstration</summary>

Simulator Screen Recording - Iphone 13 mini - 2024-09-16 at 19 36 59

</details>

### Logs

<details open><summary>Logs</summary>
</details>

### Flutter Doctor output

<details open><summary>Doctor output</summary>

```console
[✓] Flutter (Channel stable, 3.22.2, on macOS 13.6 22G120 darwin-arm64, locale en-BR)
    • Flutter version 3.22.2 on channel stable at /Users/ramonpessoal/Desktop/Source/flutter_sdk/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 761747bfc5 (3 months ago), 2024-06-05 22:15:13 +0200
    • Engine revision edd8546116
    • Dart version 3.4.3
    • DevTools version 2.34.3

[✗] Android toolchain - develop for Android devices
    ✗ Unable to locate Android SDK.
      Install Android Studio from: https://developer.android.com/studio/index.html
      On first launch it will assist you in installing the Android SDK components.
      (or visit https://flutter.dev/docs/get-started/install/macos#android-setup for detailed instructions).
      If the Android SDK has been installed to a custom location, please use
      `flutter config --android-sdk` to update to that location.


[✓] Xcode - develop for iOS and macOS (Xcode 15.2)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 15C500b
    • CocoaPods version 1.15.2

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2024.1)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 17.0.10+0-17.0.10b1087.21-11609105)

[✓] IntelliJ IDEA Community Edition (version 2021.3.3)
    • IntelliJ at /Applications/IntelliJ IDEA CE.app
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart

[✓] VS Code (version 1.93.1)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.96.0

[✓] Connected device (4 available)
    • Iphone 13 mini (mobile)         • E31E1A05-41A6-4979-A06D-3D7D83C1D62C • ios            • com.apple.CoreSimulator.SimRuntime.iOS-17-2 (simulator)
    • macOS (desktop)                 • macos                                • darwin-arm64   • macOS 13.6 22G120 darwin-arm64
    • Mac Designed for iPad (desktop) • mac-designed-for-ipad                • darwin         • macOS 13.6 22G120 darwin-arm64
    • Chrome (web)                    • chrome                               • web-javascript • Google Chrome 128.0.6613.138

[✓] Network resources
    • All expected network resources are available.

! Doctor found issues in 1 category.
```

</details>

### PR DESCRIPTION
**Pull request by RamonFarizel**
Add  `magnificationScale `to `CupertinoMagnifier` for Zoom Effect

The CupertinoMagnifier widget is not displaying the expected zoom effect.
This issue arises because the widget internally wraps the RawMagnifier, which has its magnificationScale attribute set to the default value of 1. As a result, no magnification is applied, and the zoom effect is absent.

Fixes #155275 

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [ ] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] I followed the [breaking change policy] and added [Data Driven Fixes] where supported.
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/blob/main/docs/contributing/Tree-hygiene.md#overview
[Tree Hygiene]: https://github.com/flutter/flutter/blob/main/docs/contributing/Tree-hygiene.md
[test-exempt]: https://github.com/flutter/flutter/blob/main/docs/contributing/Tree-hygiene.md#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/blob/main/docs/contributing/Style-guide-for-Flutter-repo.md
[Features we expect every widget to implement]: https://github.com/flutter/flutter/blob/main/docs/contributing/Style-guide-for-Flutter-repo.md#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/blob/main/docs/contributing/Tree-hygiene.md#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/blob/main/docs/contributing/Chat.md
[Data Driven Fixes]: https://github.com/flutter/flutter/blob/main/docs/contributing/Data-driven-Fixes.md

### CODE REVIEW

File: packages/flutter/lib/src/cupertino/magnifier.dart [hunk 1]
```diff
@@ -258,7 +258,9 @@ class CupertinoMagnifier extends StatelessWidget {
     this.borderSide =
         const BorderSide(color: Color.fromARGB(255, 232, 232, 232)),
     this.inOutAnimation,
-  });
+    this.magnificationScale = 1,
+  }) : assert(magnificationScale != 0,
```
Comment (navaronbracke) [inline packages/flutter/lib/src/cupertino/magnifier.dart:262]: Don't compare against zero. Negative values should also be avoided.

```suggestion
  }) : assert(magnificationScale > 0, 'The magnification scale should be greater than zero.');
```

File: packages/flutter/lib/src/cupertino/magnifier.dart [hunk 2]
```diff
@@ -334,6 +336,11 @@ class CupertinoMagnifier extends StatelessWidget {
   /// point offset defined in [kMagnifierAboveFocalPoint].
   final Offset additionalFocalPointOffset;
 
+  /// How "zoomed in" the magnification subject is in the lens.
```
Comment (navaronbracke) [inline packages/flutter/lib/src/cupertino/magnifier.dart:339]:
```suggestion
  /// The magnification scale for the magnifier.
  ///
  /// Defaults to 1.0, which indicates that the magnifier does not apply any magnification.
```

File: packages/flutter/lib/src/cupertino/magnifier.dart [hunk 3]
```diff
@@ -258,7 +258,9 @@ class CupertinoMagnifier extends StatelessWidget {
     this.borderSide =
         const BorderSide(color: Color.fromARGB(255, 232, 232, 232)),
     this.inOutAnimation,
-  });
+    this.magnificationScale = 1,
```
Comment (navaronbracke) [inline packages/flutter/lib/src/cupertino/magnifier.dart:261]: This change is going to need a test. You'll need the following tests:
1. A test that verifies that using zero or negative scales throws an assertion error with the correct message. Just trying to create a CupertinoMaginifier and using the correct throws matcher should work.
2. A test that checks that a scale of 1.0 did not scale the child
3. A test that checks that a scale (that is different from 1.0) scaled the child properly.

For 2 and 3 you might be able to check if the RawMagnifier received thew correct scale from the CupertinoMagnifier.

Comment (navaronbracke) [inline packages/flutter/lib/src/cupertino/magnifier.dart:261]: Which part of the actual magnifier widget uses this parameter? Currently it looks like it is still unused. Or is this PR still a work in progress?

Comment (RamonFarizel) [reply packages/flutter/lib/src/cupertino/magnifier.dart]: I forgot to push it. My bad.

File: packages/flutter/lib/src/cupertino/magnifier.dart [hunk 4]
```diff
@@ -334,6 +336,13 @@ class CupertinoMagnifier extends StatelessWidget {
   /// point offset defined in [kMagnifierAboveFocalPoint].
   final Offset additionalFocalPointOffset;
 
+  /// The magnification scale for the magnifier.
+  ///
+  /// Defaults to 1.0, which indicates that the magnifier does not apply any magnification.
+  ///
+  /// The default is 1.0, which is no magnification.
```
Comment (navaronbracke) [inline packages/flutter/lib/src/cupertino/magnifier.dart:343]: This line is redundant with the statement above, so it can be removed.

Comment (RamonFarizel) [reply packages/flutter/lib/src/cupertino/magnifier.dart]: I recently updated the master branch in my local environment, and now I'm encountering several errors. I’ve tried using `flutter update-packages`, but that didn’t resolve the issue. This is blocking me from starting to write tests.

Could you please assist me with this? Alternatively, could you direct me to the best resources or Discord channels for getting help with Flutter environment issues?

Thanks in advance!

File: packages/flutter/test/cupertino/magnifier_test.dart [hunk 5]
```diff
@@ -263,5 +263,64 @@ void main() {
...
+            home: Scaffold(body: CupertinoMagnifier()),
+          ),
+        );
+
+        expect(
+          tester.widget(find.byType(RawMagnifier)),
+          isA<RawMagnifier>().having((RawMagnifier t) => t.magnificationScale,
+              'magnificationScale', 1),
+        );
+      });
```
Comment (nate-thegrate) [inline packages/flutter/test/cupertino/magnifier_test.dart:305]:
```suggestion
      testWidgets('CupertinoMagnifier magnification scale defaults to 1', (WidgetTester tester) async {
        await tester.pumpWidget(
          const MaterialApp(
            home: Scaffold(body: CupertinoMagnifier()),
          ),
        );

        expect(
          tester.widget(find.byType(RawMagnifier)),
          isA<RawMagnifier>().having(
            (RawMagnifier t) => t.magnificationScale,
            'magnificationScale',
            1,
          ),
        );
      });
```

File: packages/flutter/test/cupertino/magnifier_test.dart [hunk 6]
```diff
@@ -263,5 +263,64 @@ void main() {
...
+          ),
+        );
+
+        expect(
+          tester.widget(find.byType(RawMagnifier)),
+          isA<RawMagnifier>().having((RawMagnifier t) => t.magnificationScale,
+              'magnificationScale', 2),
+        );
+      });
+    });
```
Comment (nate-thegrate) [inline packages/flutter/test/cupertino/magnifier_test.dart:324]:
```suggestion

      testWidgets('Magnification scale argument is passed to the RawMagnifier', (WidgetTester tester) async {
        await tester.pumpWidget(
          const MaterialApp(
            home: Scaffold(
              body: CupertinoMagnifier(
                magnificationScale: 2,
              ),
            ),
          ),
        );

        expect(
          tester.widget(find.byType(RawMagnifier)),
          isA<RawMagnifier>().having(
            (RawMagnifier t) => t.magnificationScale,
            'magnificationScale',
            2,
          ),
        );
      });
```

**Review by nate-thegrate — APPROVED** [review]
LGTM, pending some whitespace/wording changes to the tests.
(Not your fault, by the way! It looks like a bunch of tests in this file have the same issues.)

Thanks for the contribution!

**Comment by RamonFarizel** [comment]
> LGTM, pending some whitespace/wording changes to the tests. (Not your fault, by the way! It looks like a bunch of tests in this file have the same issues.)
> 
> Thanks for the contribution!

Thank you, Nate!
With those contributions, I expect to overcome this language barrier.
By the way, is there any discord channel recommendation that I should check more frequently?
I want to keep in touch with the community but I must confess that get a bit lost seeing so many channels.

**Comment by nate-thegrate** [comment]
No worries!

I'm sure everyone's experience is different, but here are some channels that I find myself interacting with pretty often:
- `#general`: general chat for people who make Flutter apps
- `#help`: for giving/receiving help for things related to making Flutter apps
- `#hackers-💭`: general chat for people who contribute to Flutter
(it's also where we request [test exemptions](https://github.com/flutter/flutter/blob/main/docs/contributing/Tree-hygiene.md#tests))

**Review by justinmc — APPROVED** [review]
LGTM 👍 

@RamonFarizel Does the latest native iOS apply a magnification by default nowadays? Maybe we should update the default magnification as well if so (possibly in a separate PR).

**Comment by RamonFarizel** [comment]
> LGTM 👍
> 
> @RamonFarizel Does the latest native iOS apply a magnification by default nowadays? Maybe we should update the default magnification as well if so (possibly in a separate PR).

Did you mean updating the `magnificationScale` value to 1.5 or 2.0 by default? 
I searched the ios docs but couldn't find their default value.

**Pull request merged**